README.md

```diff
@@ -102,13 +102,7 @@ For this section, you should address the following:
 
 * An experiment that verifies that your implementations of `lin` use no more time than O(n+m) to find all occurrences of a given pattern in a text. Remember to explain your choice of test data. What are “best” and “worst” case inputs?
 
-You can insert pictures here like this:
+The experiment showing the time complexity of both algorithms are shown in the figure below. The running time for the linear algorithm is linear and the running time for the naive algorithm is quadratic as supposed. To begin with I used the test data where I did not see the supposed running times. Therefore, I made some longer sequences and patterns and set up a worst case scenario. The data used to make the figure is found in src/test_data.
```

Comment:
This figure does not convince me of the running times. It is generally a bad idea to say that a figure shows something unless you explain *how* it shows it, and I don't see it here. What should I expect to see when plotting $O(n+m)$ over $n\cdot m$, for example? And how does the figure show the right complexity? For $O(nm)$ over $n\cdot m$, I would expect a line, so that is what the blue dots show should, but they don't unless you remove two of them as outliers.

I am not claiming here that our implementations don't have the correct running times, but it would require more than this figure to convince me that it is true.

For that, you also want to consider *when* the naive algorithm would hit a $O(nm)$ running time. Is that the expected running time with the random test data? If not, you are not demonstrating the worst-case complexity (since you are not using worst-case data). What "worst-case" and "best-case" data looks like was part of the question here, and I don't see that addressed at all.